src/audio/module_adapter/module_adapter.c

```diff
@@ -622,12 +622,6 @@ module_single_sink_setup(struct comp_dev *dev,
 	int i = 0;
 
 	list_for_item(blist, &dev->bsource_list) {
-		/* check if the source dev is in the same state as the dev */
```

Comment:
@lyakh this is a special case for the mixin/mixout alone right or is it widely applicable for any connected sink pipeline because of the order in which trigger pipelines with IPC4?

Reply:
@ranj063 I'm not sure. Definitely it's only possibly with module-adapter components. In our "popular" topologies I only see mixin-out, copier and volume being on the border between two pipelines. Copier isn't using module-adapter, but we could check some volume pipelines. I so far haven't figured out where in volume consumed and produced bytes are updated, will try to figure that out tomorrow. Then I'll add some debugging there and check.

Reply:
@ranj063 I tested with a capture pipeline like DMIC -> gain.A -> copier.A -> gain.B -> PCM and I don't see any difference with or without this PR so at least it isn't breaking that use-case either

Reply:
@ranj063 I know why there was no difference. The difference is that although in one case `module_process()` is called from `module_adapter_simple_copy()` with `num_input_buffers` of `num_output_buffers` equal to 0, both those parameters are ignored by `volume_process()`... If I add there something like
```
	if (!num_output_buffers || !num_input_buffers)
		return 0;
```
then of course the behaviour changes. But still both possibilities seem to work. Maybe it's because with capture usually you don't have to send data at a constant rate. And also the effect of this PR seems to depend strongly on individual processing modules - how they use those parameters.

Reply:
Ack, I don't think this can work if the adapter makes this call. I think it should be component that handles this logic, and indeed, if it has data in its buffers, it is ACTIVE and there's room in outputs, processing should happen.